ui/conductor/src/traits/lighting/useLightingTrait.js

```diff
@@ -1,89 +1,133 @@
+  const brightnessLevelNumber = computed(() => light.value?.levelPercent);
+
+  const brightnessLevelString = computed(() => {
+    if (brightnessLevelNumber.value === 0) {
+      return 'Off';
+    } else if (brightnessLevelNumber.value === 100) {
+      return 'Max';
+    } else if (brightnessLevelNumber.value > 0 && brightnessLevelNumber.value < 100) {
+      return `${brightnessLevelNumber.value.toFixed(0)}%`;
+    }
+
+    return '';
+  });
+
+  const lightIcon = computed(() => {
+    if (brightnessLevelNumber.value === 0) {
+      return 'mdi-lightbulb-outline';
+    } else if (brightnessLevelNumber.value > 0) {
+      return 'mdi-lightbulb-on';
+    } else {
+      return '';
+    }
+  });
+
+
+  // ---------------- Light Control ---------------- //
   /**
-   * @param {number|Brightness.AsObject|UpdateBrightnessRequest.AsObject} req
+   * @param {number|Brightness.AsObject|UpdateBrightnessRequest.AsObject} request
+   * @return {void}
    */
-  function doUpdateBrightness(req) {
-    if (typeof req === 'number') {
-      req = {levelPercent: Math.min(100, Math.round(req))};
+  const doUpdateBrightness = (request) => {
+    if (typeof request === 'number') {
+      request = {
+        levelPercent: Math.min(100, Math.round(request))
+      };
     }
-    if (!req.hasOwnProperty('brightness')) {
-      req = {brightness: req};
+    if (!request.hasOwnProperty('brightness')) {
+      request = {
+        brightness: request
+      };
     }
-    req.name = props.name;
-    updateBrightness(req, updateValue);
-  }
+    request.name = toValue(queryObject).name;
 
-  //
-  //
-  // Watch
-  // Depending on paused state/device name, we close/open data stream(s)
-  watch(
-      [() => props.paused, () => props.name],
-      ([newPaused, newName], [oldPaused, oldName]) => {
-        // only for LightSensor
-        if (newPaused === oldPaused && newName === oldName) return;
-
-        if (newPaused) {
-          closeResource(lightValue);
-        }
-
-        if (!newPaused && (oldPaused || newName !== oldName)) {
-          closeResource(lightValue);
-          pullBrightness({name: newName}, lightValue);
-        }
-      },
-      {immediate: true, deep: true, flush: 'sync'}
-  );
+    updateBrightness(request, lightUpdate);
+  };
 
-  //
-  //
-  // UI error handling
-  let unwatchUpdateError;
-  onMounted(() => {
-    unwatchUpdateError = errorStore.registerTracker(updateValue);
+
+  // ----------- Errors ----------- //
+  const lightError = computed(() => {
+    return light.streamError || lightUpdate.error;
   });
-  onUnmounted(() => {
-    closeResource(lightValue);
-    if (unwatchUpdateError) unwatchUpdateError();
+
+
+  // ----------- Loading ----------- //
+  const lightLoading = computed(() => {
+    return light.loading || lightUpdate.loading;
   });
 
+  // ---------------- Return ---------------- //
   return {
-    lightValue,
-    updateValue,
-    doUpdateBrightness
+    light,
+    lightUpdate,
```

Comment:
Can these be named after the resource not the trait? With traits that have multiple resources this will get confusing.